File: frontend/src/types/workflow.ts
/**
 * Workflow related interfaces
 */

// Define workflow interface
export interface Workflow {
  namespaceID: string;
  workflowName: string;
  workflowType: string;
  sourceBucketARN: string;
  destBucketARN: string;
  sourceRoleARN: string;
  destRoleARN: string;
  sourceAccountNumber: string;
  destAccountNumber: string;
  sourceRegion: string;
  destRegion: string;
  status: string;
  createdAt?: string;
  startedAt?: string;
  backfillCompletedAt?: string;
  completedAt?: string;
  runtimeConfig?: RuntimeConfig;
}

export interface RuntimeConfig {
  dashboardUrl?: string;
  isReplicationTimeControlEnabled?: boolean;
}

export interface CreateWorkflowRequest {
  sourceBucketName: string;
  destBucketName: string;
  sourceAccountNumber: string;
  destAccountNumber: string;
  sourceRegion: string;
  destRegion: string;
  runtimeConfig: {
    isReplicationTimeControlEnabled: boolean;
  };
}

export interface WorkflowView {
  namespaceID: string;
  workflowName: string;
  sourceBucketName: string;
  destBucketName: string;
  sourceAccountNumber: string;
  destAccountNumber: string;
  sourceRegion: string;
  destRegion: string;
  sourceRoleARN: string;
  destRoleARN: string;
  status: string;
  createdAt?: string;
  startedAt?: string;
  backfillCompletedAt?: string;
  completedAt?: string;
  runtimeConfig?: {
    dashboardUrl?: string;
    isReplicationTimeControlEnabled?: boolean;
  };
}

export interface GetWorkflowResponse {
  workflow: Workflow;
}

export interface ListWorkflowsResponse {
  workflows: Workflow[];
}

File: src/main/java/com/amazon/tdm/s3a/model/Workflow.java
package com.amazon.tdm.s3a.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

/**
 * Workflow details
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Workflow {
    /**
     * ID being provided by createWorkflow API to S3A
     */
    private String namespaceID;
    
    /**
     * Name of the Workflow
     */
    private String workflowName;
    
    /**
     * Type of workflow
     */
    private String workflowType;
    
    /**
     * Status of the Workflow
     */
    private String status;
    
    private String sourceBucketARN;
    private String destBucketARN;
    private String sourceRoleARN;
    private String destRoleARN;
    private Map<String, String> workflowConfig;
    private RuntimeConfig runtimeConfig;
    private String sourceAccountNumber;
    private String destAccountNumber;
    private String sourceRegion;
    private String destRegion;
    private Map<String, String> sourceBucketConfig;
    private Map<String, String> destBucketConfig;
    private String bopsJobID;
    private String state;
    private String ackedNotification;
    private String sentNotification;
    private String jobReportBucketARN;
    
    // Metadata
    private String createdAt;
    private String startedAt;
    private String backfillCompletedAt;
    private String completedAt;
}